src/lib/blocks.js

```diff
@@ -158,6 +158,16 @@ export default function (vm, useCatBlocks) {
     ScratchBlocks.Blocks.sound_sounds_menu.init = function () {
         const json = jsonForMenuBlock('SOUND_MENU', soundsMenu, soundColors, []);
         this.jsonInit(json);
+        this.inputList[0].removeField('SOUND_MENU');
+        this.inputList[0].appendField(new ScratchBlocks.FieldDropdown(() => {
+            return soundsMenu();
+        }, (newValue) => {
+            if (newValue === 'SOUND_RECORD') {
+                  ScratchBlocks.recordSoundCallback();
+                  return null;
```

Comment:
Curious why you're returning `null` here. That represents an invalid value, which reverts to the previous value and may be what you want. I'm just not sure what the user-facing behavior is.

Reply:
Yeah that is. If you try this in Scratch (choose "record..." from a sound picker menu) it brings up a modal to do the recording, then dumps you off to the sound editor tab, and when you come back the sound that was originally selected still is.